src/preact/amp-base-element.js

```diff
@@ -0,0 +1,119 @@
+import {ActionTrust_Enum} from '#core/constants/action-constants';
+import {dict} from '#core/types/object';
+
+import {PreactBaseElement} from './base-element';
+
+export class AmpPreactBaseElement extends PreactBaseElement {
+  // todo(kvchari): confirm this can be safely moved
+  /** @override */
+  mutatedAttributesCallback() {
+    // eslint-disable-next-line local/restrict-this-access
+    if (this.container_) {
+      // eslint-disable-next-line local/restrict-this-access
+      this.scheduleRender_();
+    }
+  }
+
+  // todo(kvchari): confirm this can be safely moved
+  /** @override */
+  attemptChangeHeight(newHeight) {
+    return super.attemptChangeHeight(newHeight).catch((e) => {
+      // It's okay to disable this lint rule since we check that the restricted
+      // method exists.
+      // eslint-disable-next-line local/restrict-this-access
+      if (this.getOverflowElement && !this.getOverflowElement()) {
+        console./* OK */ warn(
+          '[overflow] element not found. Provide one to enable resizing to full contents.',
+          this.element
+        );
+      }
+      throw e;
+    });
+  }
+
+  /** @override */
+  getDefaultProps() {
+    return dict({
+      // eslint-disable-next-line local/dict-string-keys
```

Comment:
Why are these keys strings?

Reply:
Honestly no idea - the existing `defaultProps_` is typed as a `JsonObject` which has a lint rule that it must be constructed with object literals with string props, which means you can't use spread `...{}` syntax. But I want to use that syntax to "inherit" the parent class' `defaultProps_`, so I have to disable the rule. Not sure what the preferred solution is here: disable the lint rule, change the type of the `defaultProps_` obj, etc.

Reply:
Given Alan's PR, can skip `dict()` entirely

Reply:
removed